tests/unittest/_torch/attention/backend_case.py

```
@@ -0,0 +1,812 @@
+    def kv_new_lens(self) -> List[int]:
+        """New KV tokens per request (encoder side for cross; == seq_lens for self)."""
+        return self.seq_lens_kv if self.seq_lens_kv is not None else self.seq_lens
+
+    @property
+    def nnz_kv(self) -> int:
+        return sum(self.kv_new_lens)
+
+    @property
+    def token_nums(self) -> List[int]:
+        # Total KV tokens per request after the backend appends the new KV.
+        return [c + s for c, s in zip(self.num_cached_tokens, self.kv_new_lens)]
```

Ruff (0.15.18)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)